mesonbuild/modules/icestorm.py

```diff
@@ -34,17 +34,19 @@ class ProjectKwargs(TypedDict):
         sources: T.List[T.Union[mesonlib.FileOrString, build.GeneratedTypes]]
         constraint_file: T.Union[mesonlib.FileOrString, build.GeneratedTypes]
 
-class IceStormModule(ExtensionModule):
+class IceStormModule(NewExtensionModule):
```

Comment:
Do not update this to the new extension module API. The purpose of the new extension module API is specifically, solely, exclusively, to drop access to self.interpreter... which you have added here in the `__init__` method for IceStormModule instead?

It is correct to share implementation of assigning `self.interpreter`.